linkerd/docs/protocol-http.md

```diff
@@ -61,6 +64,59 @@ concurrent usage of linkerd. Changing these parameters may
 significantly alter linkerd's performance characteristics.
 </aside>
 
+
+<a name="http-1-1-server"></a>
+## HTTP Servers ##
+
+HTTP servers accept additional configuration parameters.
+
+> Example: default
+```
+addForwardedHeader: {}
+```
+
+Key | Default Value | Description
+--- | ------------- | -----------
+addForwardedHeader | null | If set, a `Forwarded` header is added to all requests.  See [below](#http-1-1-forwarded).
+
+<a name="http-1-1-forwarded"></a>
+### Adding the `Forwarded` header ###
+
+[RFC 7239](https://tools.ietf.org/html/rfc7239) describes how a
+`Forwarded` header may be added to requests by proxies. This RFC
+requests that this header not be added unless explicitly configured
+and that proxies obfuscate IP addresses unless explicitly configured
+to transmit them.
+
+Key | Default Value | Description
+--- | ------------- | -----------
+by  | `kind: random` | The [labeler](#http-1-1-forwarded-labeler) to use with the router's server address
+for | `kind: random` | The [labeler](#http-1-1-forwarded-labeler) to use with the upstream client's address
+
+<a name="http-1-1-forwarded-labelers"></a>
+#### Endpoint labelers ####
+
+The `Forwarded` header includes labels describing the endpoints of the
+upstream connection. Because this is sensitive information, it is
+typically randomized.
+
+> Example
+```
+addForwardedHeader:
+  for: {kind: ip}
+  by:
+    kind: static
+    label: linkerd
+```
+
+Kind | Description
+---- |
+ip | A textual IP address like _192.168.1.1_ or _"[2001:db8:cafe::17]"_.
+ip:port | A textual IP:PORT address like _"192.168.1.1:80"_ or _"[2001:db8:cafe::17]:80"_.
+**random** | Generate an obfuscated random label like __6Oq8jJ_.
```

Comment:
The spec says that, for the purposes of correlation, a random id can be used for the lifetime of the connection, which is how I implemented this in linkerd. I think it's to be used to show the _existence of_ an intermediary and not any additional diagnostic information. We should probably trace these values (and even store them as records?)